crates/api_models/src/payments.rs

```diff
@@ -1040,56 +1084,87 @@ pub struct GpaySessionTokenData {
 #[serde(tag = "wallet_name")]
 #[serde(rename_all = "lowercase")]
 pub enum SessionToken {
+    /// The session response structure for gpay
     Gpay {
+        /// The data gpay requires
         #[serde(flatten)]
         data: GpayMetadata,
+        /// The transaction info gpay requires
         transaction_info: GpayTransactionInfo,
     },
+    /// The session response structure for klarna
     Klarna {
+        /// The session token for klarna
         session_token: String,
+        /// The identifier for the session
         session_id: String,
     },
+    /// The session response structure for paypal
     Paypal {
+        /// The session token for paypal
         session_token: String,
     },
+    /// The session response structure for applepay
     Applepay {
+        /// Timestamp at which session is requested
         epoch_timestamp: u64,
+        /// Timestamp at which session expires
         expires_at: u64,
+        /// The identifier for the merchant session
         merchant_session_identifier: String,
+        /// Applepay generates unique ID (UUID) value
         nonce: String,
+        /// The identifier for the merchant
         merchant_identifier: String,
+        /// The domain name of the merchant which is registered in applepay
         domain_name: String,
+        /// The name to be displayed on applepay button
         display_name: String,
+        /// A string which represents the properties of a payment
         signature: String,
+        /// The identifier for the operational analytics
         operational_analytics_identifier: String,
+        /// The number of retries to get the session repsonse
         retries: u8,
+        /// The identifier for the connector transaction
         psp_id: String,
     },
 }
 
-#[derive(Default, Debug, serde::Serialize, Clone)]
+#[derive(Default, Debug, serde::Serialize, Clone, ToSchema)]
 pub struct PaymentsSessionResponse {
+    /// The identifier for the payment
     pub payment_id: String,
+    /// This is a 15 minute expiry token which shall be used from the client to authenticate and perform sessions from the SDK
     pub client_secret: Secret<String, pii::ClientSecret>,
```

Comment:
1. Please update the doc comment here as well.
2. ```suggestion
       #[schema(value_type = String)]
       pub client_secret: Secret<String, pii::ClientSecret>,
   ```